### ISSUE TITLE
Support StarScream V4+

### ISSUE THREAD

**Issue by PandaZhang530** [WalletConnect/WalletConnectSwiftV2#1055]
<img width="445" alt="image" src="https://github.com/WalletConnect/WalletConnectSwiftV2/assets/140152146/9300bf2a-24fc-4652-b9d9-c46fc64206be">

**Comment by AndrewPetrov** [comment]
I use my own implementation for StarScream V4

**Comment by PandaZhang530** [comment]
> I use my own implementation for StarScream V4
> 
> ```
> class MyWebSocket: Starscream.WebSocket, WebSocketConnecting {
>   private var _isConnected = true
> 
>   public var isConnected: Bool {
>     _isConnected
>   }
> 
>  var onConnect: (() -> Void)?
> 
>  var onDisconnect: ((Error?) -> Void)?
> 
>  var onText: ((String) -> Void)?
> 
>   convenience init(newRequest: URLRequest) {
>     self.init(request: newRequest, useCustomEngine: false)
> 
>     onEvent = { [weak self] event in
>       guard let self else { return }
> 
>       switch event {
>       case .connected:
>         _isConnected = true
>         onConnect?()
>       case .disconnected(let reason, let code):
>         _isConnected = false
>         onDisconnect?(NSError(domain: reason, code: Int(code), userInfo: nil))
>       case .text(let text):
>         onText?(text)
>       case .binary:
>         break
>       case .pong:
>         break
>       case .ping:
>         break
>       case .error(let error):
>         onDisconnect?(error)
>       case .viabilityChanged:
>         break
>       case .reconnectSuggested:
>         break
>       case .cancelled:
>         _isConnected = false
>       @unknown default:
>         break
>       }
>     }
>   }
> }
> ```
Thanks a lot, that does work~~

**Comment by simonmcl** [comment]
@AndriiPetrovDev I've tried updating to Starscream 4.0.8 (currently the latest) and using this code you provided, but this version of Starscream seems to be incredibly unstable, crashing very frequently. I updated hoping to solve a much rarer crash using 3.18 but this is a huge step backwards. What has your experience been? For now I'll downgrade, but will likely investigate doing it natively again, as this has been a significant pain point for us
